plugins/performance-lab/includes/admin/rest-api.php

```diff
@@ -0,0 +1,201 @@
+<?php
+/**
+ * REST API integration for the plugin.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Namespace for performance-lab REST API.
+ *
+ * @var string
+ */
+const PERFLAB_REST_API_NAMESPACE = 'performance-lab/v1';
+
+/**
+ * Route for activating plugin.
+ *
+ * @var string
+ */
+const PERFLAB_ACTIVATE_PLUGIN_ROUTE = '/activate-plugin';
+
+/**
+ * Route for fetching plugin settings URL.
+ *
+ * @var string
+ */
+const PERFLAB_PLUGIN_SETTINGS_URL_ROUTE = '/plugin-settings-url';
```

Comment:
```suggestion
const PERFLAB_PLUGIN_SETTINGS_URL_ROUTE = '/plugin-settings-url/(?P<slug>[a-z0-9_-]+)';
```

Reply:
See above, better would be:
```suggestion
const PERFLAB_PLUGIN_SETTINGS_URL_ROUTE = '/plugins/(?P<slug>[a-z0-9_-]+):settings-url';
```

This way the two routes are also consistently centered around a specific plugin.

Reply:
But in this case it's just a regular REST API endpoint. No need for any workaround for a non-RESTy `POST` request, right?

Reply:
How do you mean? How is this a workaround?

Reply:
I mean, as I understand, a workaround like this is needed when the REST API endpoint is not a noun and/or the existing HTTP methods don't make semantic sense. According to [AIP-136](https://google.aip.dev/136):

> Resource-oriented design ([AIP-121](https://google.aip.dev/121)) uses custom methods to provide a means to express arbitrary actions that are difficult to model using only the standard methods.

What we discussed above in https://github.com/WordPress/performance/pull/1646/files#r1841160336 is to add a verb to the endpoint for use with `POST`. But here for the settings URL, it's a classic REST API endpoint, like most other REST API endpoints in WordPress, where we're just `GET` a noun. So so the normal REST design makes more sense to me.

Reply:
That's a good point. Though the current naming still seems incorrect in that it's the settings URL of a specific plugin. So a more appropriate name would be:
```suggestion
const PERFLAB_PLUGIN_SETTINGS_URL_ROUTE = '/plugins/(?P<slug>[a-z0-9_-]+)/settings-url';
```

This way it is a resource, but under the plugin it belongs to. Arguably with that approach, the 404 error if there's no settings URL makes more sense too.

Reply:
See https://github.com/WordPress/performance/pull/1646#discussion_r1842717861, discard my previous comment.

Let's continue that discussion on the other thread, to center the conversation about endpoint naming in one place.